app/src/main/java/com/github/sdp/mediato/data/ReviewInteractionDatabase.java

```diff
@@ -0,0 +1,130 @@
+package com.github.sdp.mediato.data;
+
+import static com.github.sdp.mediato.data.DatabaseUtils.getReactionReference;
+
+import com.github.sdp.mediato.errorCheck.Preconditions;
+import com.github.sdp.mediato.model.Review;
+import com.github.sdp.mediato.model.post.Reaction;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This class is used to interact with the database for reviews.
+ */
+public class ReviewInteractionDatabase {
+
+    public static FirebaseDatabase database = FirebaseDatabase.getInstance();
+
+    /**
+     * Determines if the user has liked a given review.
+     * @param refUsername the username of the user who is checked
+     * @param tarUsername the username of the user who made the review
+     * @param collectionName the name of the collection
+     * @param review the review
+     * @return a completable future that returns true if the user has liked the review, false otherwise
+     */
+    public static CompletableFuture<Boolean> likes(String refUsername, String tarUsername, String collectionName, String review) {

```

Comment:
Couldn't we use a single method for likes and dislikes ? This would make the code a little smaller and add the possibility of adding other reactions type in the future, but it is not an objective point of view. Up to you to decide.

Reply:
I don't think so. It would make the method signature way too long for code climate.